sdk/Sdk/Abstractions/SdkRetryOptions.cs

```diff
@@ -5,23 +5,14 @@ namespace Microsoft.Azure.Functions.Worker.Sdk
 {
     internal class SdkRetryOptions
     {
-        public SdkRetryOptions(string strategy, int maxRetryCount, string? delayInterval, string? minimumInterval, string? maximumInterval)
-        {
-            Strategy = strategy;
-            MaxRetryCount = maxRetryCount;
-            DelayInterval = delayInterval;
-            MinimumInterval = minimumInterval;
-            MaximumInterval = maximumInterval;
-        }
+        public string? Strategy { get; set; }
```

Comment:
A refactor to match [SdkFunctionMetadata](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/sdk/Sdk/Abstractions/SdkFunctionMetadata.cs). In previous design there were empty retry options unexpectedly in some tests I ran (both unit and manual e2e) trying to fix the failing `functions.metadata` generation file. I believe this is the correct pattern - it matches official docs for json serialization and is passing all the tests.